editor/editor_run.cpp

```diff
@@ -184,9 +184,44 @@ Error EditorRun::run(const String &p_scene) {
 		args.push_back(p_scene);
 	}
 
-	String exec = OS::get_singleton()->get_executable_path();
+	int instances = EditorSettings::get_singleton()->get_project_metadata("debug_options", "run_debug_instances", 1);
+	for (int instance_num = 1; instance_num <= instances; instance_num++) {
+		List<String> instance_args;
+		for (int i = 0; i < args.size(); i++) {
+			instance_args.push_back(args[i]);
+		}
+
+		String exec;
+		String instance_arg_setting = "editor/run/instance_" + itos(instance_num) + "_run_args";
+		String raw_custom_args = ProjectSettings::get_singleton()->get(instance_arg_setting);
+		if (raw_custom_args.is_empty()) {
+			// Fallback to main run args if no instance specific run args exist.
+			raw_custom_args = ProjectSettings::get_singleton()->get("editor/run/main_run_args");
+		}
+		parse_run_args(&exec, &instance_args, raw_custom_args);
+
+		printf("Running: %s", exec.utf8().get_data());
+		for (const String &E : instance_args) {
+			printf(" %s", E.utf8().get_data());
+		};
+		printf("\n");
+
+		OS::ProcessID pid = 0;
+		Error err = OS::get_singleton()->create_instance(instance_args, &pid);
+		ERR_FAIL_COND_V(err, err);
+		pids.push_back(pid);
+	}
+
+	status = STATUS_PLAY;
+	if (!p_scene.is_empty()) {
+		running_scene = p_scene;
+	}
+
+	return OK;
+}
 
-	const String raw_custom_args = ProjectSettings::get_singleton()->get("editor/run/main_run_args");
+void EditorRun::parse_run_args(String *exec_path, List<String> *args, const String raw_custom_args) {
```

Comment:
This fits more the godot codebase:

```suggestion
void EditorRun::parse_run_args(String &r_exec_path, List<String> &r_args, const String p_raw_custom_args) {
```